generator.go

```diff
@@ -193,6 +194,32 @@ func NewGenWithOptions(opts ...GenOption) *Gen {
 	return gen
 }
 
+// MonotonicGen extends the Gen struct with a counter for batch generation.
+//
+// MonotonicGen ensures the generation of strictly monotonic UUIDs within a
+// batch by utilizing a counter in conjunction with timestamps. This is
+// particularly useful for applications requiring ordered identifiers, such
```

Comment:
> for applications requiring ordered identifiers

v6 and v7 UUIDs are already ordered so none of this is strictly _necessary_ except to enable batch generation, and I'm not sold on the utility there.  I have used this library to generate UUIDs on the order of tens of millions per second sustained over 1000s of nodes in a distributed system without ever running into a scenario where I wanted/needed to pre-allocate a block of values.

Reply:
I agree with @dylan-bourque - I'm really not sure how useful this is relative to the increased complexity. I haven't generated millions per second, but I can't see any reason that the existing implementation doesn't do what you want. Moreover, I'm concerned that having two separate implementations of generating a new UUIDv7 (the existing public method for generating one at a time and your new private method that is called in the loop) will lead to maintenance headaches.

Is there a specific situation you have encountered in which the existing implementation fails to provide monotonically increasing values? If there is, please explain in more detail to help us understand the need.